### ISSUE TITLE
Error while using mlpack::data::Load

### ISSUE THREAD

**Issue by Lakshya1605** [mlpack/mlpack#844]
May be i am doing something wrong but whenever i am trying to load a ".csv" file into a vector i get error like below
 "error: Mat::init(): requested size is not compatible with column vector layout
terminate called after throwing an instance of 'std::logic_error'
  what():  Mat::init(): requested size is not compatible with column vector layout
Aborted (core dumped)".
But armadillo load  (ie=> x.load("***.csv") is working fine.
Is this a specified behaviour or a bug.
Thank You

**Comment by Lakshya1605** [comment]
If it is a bug i would like to work on it please can someone help me out( i

**Comment by mohit-raj** [comment]
Are you executing load through C++ Api or mlpack-CLI(command line interface) ?

**Comment by mohit-raj** [comment]
In case you are load armadillo then you need to transpose the matrix after load, if the coordinate list is in row major format. It might be the cause as the error says requested size is not compatible with column vector layout.
Commands to run :
Let x be the matrix name.
arma :: sp_mat x; // only if in case of loading sparse matrix (this is optional)
x.load("***.csv")  //armadilo load 
x = x.t()  // Here is your command to transpose the matrix.
Try this if it works out fine.

**Comment by Lakshya1605** [comment]
@mohit-raj  sorry my bad i missed the phrase "using mlpack::data::Load" .However i have edited it now.And yes using armadillo load is working fine  but 
Thank You

**Comment by Lakshya1605** [comment]
@mohit-raj  sorry my bad i missed the phrase "using mlpack::data::Load" .However i have edited it now.And yes using armadillo load is working fine but loading using mlpack load isn't.
Thank You

**Comment by Lakshya1605** [comment]
@mohit-raj @rcurtin  could you please confirm if it is a bug .If yes i would like to work on it ,please can you help me get started.

**Comment by rcurtin** [comment]
`data::Load()` was not intended to load column or row vectors.  You are free to help extend the function so that it does, but please be certain before you open a PR that the code compiles, is tested, and follows the mlpack style guidelines.

**Comment by Lakshya1605** [comment]
@rcurtin can you please tell me on how to get started ,where can i find the code for data::Load , and how to compile it after making changes and coding standards of mlpack ( i have tried finding the solution to the  above in the main website and also in the github page but could not find it).Sorry if i have overlooked something.
Thank You

**Comment by rcurtin** [comment]
Use a search tool like grep or something to find the code you're looking for or just poke around the repository to see where it is included from.  Once you make a change, build the library and run the tests.  There are directions for how to do this in the readme and the tutorials.

**Comment by Lakshya1605** [comment]
Thank you @rcurtin for the help i have made changes in the development version( that is i have cloned the github repository and have made changes to it)  but i am unaware on how to include this development version in compiling the program .Is it  customary to simply add the development projects header directory to my project's header search path or is their any other way.

**Comment by zoq** [comment]
Hello, you can just link against the mlpack header and library:

```
 g++ test.cpp -o test -std=c++11 -I/home/marcus/mlpack/build/include/ -L/home/marcus/mlpack/build/ -larmadillo -lmlpack
```
You might also need to link against:

```
-lboost_serialization -lboost_program_options
```

**Comment by Lakshya1605** [comment]
@rcurtin @zoq I have coded the Load function for armadillo "Column" vectors ,can you please tell me how to incorporate even "Row" vectors within the same code .Thank You!!!

**Comment by rcurtin** [comment]
Fixed with #848.
